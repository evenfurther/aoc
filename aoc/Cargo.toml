[package]
name = "aoc"
version = "0.1.0"
edition = "2021"
license = "Apache-2.0/MIT"

[dependencies]
aoc-derive = { path = "../aoc-derive" }
chrono = "0.4.31"
clap = { version = "4.4.5", features = ["derive"] }
color-eyre = "0.6.2"
eyre = "0.6.8"
itertools = "0.11.0"
lazy_static = "1.4.0"
mktemp = "0.5.1"
thiserror = "1.0.49"

[dev-dependencies]
insta = "1.33.0"
serial_test = { version = "2.0.0", default-features = false }
