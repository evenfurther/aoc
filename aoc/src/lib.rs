#![deny(clippy::pedantic)]
#![allow(clippy::missing_errors_doc, clippy::missing_panics_doc)]

pub use aoc_derive::*;

pub mod error;
pub mod input;
mod run;
mod runners;
pub mod test;

pub use run::run;
pub use runners::register_runner;
