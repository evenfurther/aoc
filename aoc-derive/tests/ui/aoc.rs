use aoc_derive::aoc;

#[aoc(day30, part1)]
fn day30part1() -> u32 {
    0
}

#[aoc(day10, part3)]
fn day10part3() -> u32 {
    0
}

#[aoc(day1, part1)]
fn day1_part1() {}

fn main() {}
