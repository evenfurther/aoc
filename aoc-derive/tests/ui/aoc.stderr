error: cannot parse day (between 1 and 25) 30
 --> tests/ui/aoc.rs:3:7
  |
3 | #[aoc(day30, part1)]
  |       ^^^^^

error: cannot parse part (1 or 2) 3
 --> tests/ui/aoc.rs:8:14
  |
8 | #[aoc(day10, part3)]
  |              ^^^^^

error: AOC part cannot return ()
  --> tests/ui/aoc.rs:14:1
   |
14 | fn day1_part1() {}
   | ^^^^^^^^^^^^^^^
